src/hooks/video/agora/index.ts

```diff
@@ -0,0 +1,233 @@
+  const { userId } = useUser();
+  const venueId = useVenueId();
+
+  const [localScreenTrack, setLocalScreenTrack] = useState<ILocalVideoTrack>();
+  const [localAudioTrack, setLocalAudioTrack] = useState<ILocalAudioTrack>();
+
+  const shareScreen = useCallback(async () => {
+    if (!client) return;
+
+    const screenTrack = await AgoraRTC.createScreenVideoTrack({}, "auto");
+
+    if (Array.isArray(screenTrack)) {
+      const [screenVideoTrack, screenAudioTrack] = screenTrack;
+      setLocalScreenTrack(screenVideoTrack);
+      setLocalAudioTrack(screenAudioTrack);
+      await client.publish(screenVideoTrack);
+      await client.publish(screenAudioTrack);
+      return;
+    }
+
+    setLocalScreenTrack(screenTrack);
+    await client.publish(screenTrack);
+  }, [client]);
+
+  const stopShare = useCallback(async () => {
+    localScreenTrack?.stop();
+    localScreenTrack?.close();
+
+    localAudioTrack?.stop();
+    localAudioTrack?.close();
+
+    await client?.unpublish();
+
+    setLocalScreenTrack(undefined);
+    setLocalAudioTrack(undefined);
+  }, [client, localAudioTrack, localScreenTrack]);
+
+  const joinChannel = async () => {
+    if (!client || !venueId || !userId) return;
+
+    const screenClientUid = await client?.join(
+      AGORA_APP_ID || "",
+      AGORA_CHANNEL || "",
+      AGORA_TOKEN || null
+    );
+
+    const experience = {
+      screenClientUid: `${screenClientUid}`,
+    };
+
+    updateTalkShowStudioExperience({ venueId, userId, experience });
+  };
+
+  const leaveChannel = useCallback(async () => {
+    stopShare();
+    await client?.leave();
+  }, [client, stopShare]);
+
+  useEffect(() => {
+    return () => {
+      leaveChannel();
+    };
+    // Otherwise, it will fire when local tracks are updated
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return {
+    localScreenTrack,
+    shareScreen,
+    stopShare,
+    joinChannel,
+    leaveChannel,
+  };
+};
+
+export const useAgoraCamera: ReactHook<
+  UseAgoraCameraProps,
+  UseAgoraCameraReturn
+> = ({ client }) => {
+  const { userId } = useUser();
+  const venueId = useVenueId();
+
+  const [localCameraTrack, setLocalCameraTrack] = useState<ILocalVideoTrack>();
+  const [
+    localMicrophoneTrack,
+    setLocalMicrophoneTrack,
+  ] = useState<ILocalAudioTrack>();
+  const { isShown: isCameraOn, setShown: setIsCameraOn } = useShowHide();
+  const {
+    isShown: isMicrophoneOn,
+    setShown: setIsMicrophoneOn,
+  } = useShowHide();
+
+  const toggleCamera = () => {
+    localCameraTrack?.setEnabled(!isCameraOn);
+    setIsCameraOn(!isCameraOn);
+  };
+
+  const toggleMicrophone = () => {
+    localMicrophoneTrack?.setEnabled(!isMicrophoneOn);
+    setIsMicrophoneOn(!isMicrophoneOn);
+  };
+
+  const joinChannel = async () => {
+    if (!client || !venueId || !userId) return;
+
+    const cameraClientUid = await client.join(
+      AGORA_APP_ID || "",
+      AGORA_CHANNEL || "",
+      AGORA_TOKEN || null
+    );
```

Comment:
Calling out usages of `AGORA_APP_ID`, `AGORA_CHANNEL`, and `AGORA_TOKEN` that are likely wrong and almost certainly will need refactoring.